src/config.ts

```diff
@@ -720,6 +754,30 @@ export async function getAsync(...target) {
     }
 }
 
+function genericSet(config, args) {
+    if (args.length < 2) {
+        throw "You must provide at least two arguments!"
+    }
+
+    const target = args.slice(0, args.length - 1)
+    const value = args[args.length - 1]
+
+    setDeepProperty(config, value, target)
+    save()
+}
+
+/** Same as [[set]], but for URLCONFIGS[pattern] instead of USERCONFIG. */
```

Comment:
Is there a way to reduce this duplication? Like, maybe we could implement the global (default) configs as a set of url-specific configurations that apply to `.*`?

Reply:
I thought about it, but how would you tell which config should be used between `.*` and `.*\.com`? We could special case `.*` and have it be the last to be checked but it wouldn't make the code prettier in my opinion.

Reply:
Ah, you're picking the longest match because it'd be a huge PITA to ask the user to maintain a list of configurations sorted by priority. Hmm. Instead of sorting by match length, sort by the length of the first capturing group? Then ".*" would have no capturing group and be at the end, and users would have a bit more control over how their regexes match if they need it.

Reply:
Oh, good idea, or we could even use the length of the pattern. This way users could trick the regex engine by adding something like `(thisisaverylongpatternthatdoesntneedtobematched)?` to their pattern in order to ensure a pattern has higher priority.

Reply:
Actually I'm not sure removing the duplication is a good idea. Users could be using different Tridactyl versions in different browser instances. Settings won't be synced anymore if we drop USERCONFIG and the user only updates one of their Tridactyls.